src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -61,6 +61,13 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
                 return GetConvertedLeftForNullCoalescingOperator(rewrittenLeft, leftConversion, rewrittenResultType);
             }
 
+            // string concatenation is never null.
+            // interpolated string lowering may introduce redundant null coalescing, which we have to remove.
+            if (IsStringConcat(rewrittenLeft))
```

Comment:
Is this still needed?

Reply:
Yes, for the case `$"{a + b}"` (where `a + b` is a `string`).